processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -105,79 +103,40 @@ public static void checkInterrupted()
   }
 
   public static final String QUERYID = "queryId";
-  private final DataSource dataSource;
+
   private final boolean descending;
   private final Map<String, Object> context;
-  private final QuerySegmentSpec querySegmentSpec;
-  private volatile Duration duration;
 
   public BaseQuery(
-      DataSource dataSource,
-      QuerySegmentSpec querySegmentSpec,
       boolean descending,
       Map<String, Object> context
   )
   {
-    Preconditions.checkNotNull(dataSource, "dataSource can't be null");
-    Preconditions.checkNotNull(querySegmentSpec, "querySegmentSpec can't be null");
-
-    this.dataSource = dataSource;
     this.context = context;
-    this.querySegmentSpec = querySegmentSpec;
     this.descending = descending;
   }
 
   @JsonProperty
   @Override
-  public DataSource getDataSource()
-  {
-    return dataSource;
-  }
-
-  @JsonProperty
-  @Override
   public boolean isDescending()
   {
     return descending;
   }
 
-  @JsonProperty("intervals")
-  public QuerySegmentSpec getQuerySegmentSpec()
-  {
-    return querySegmentSpec;
-  }
-
-  @Override
-  public Sequence<T> run(QuerySegmentWalker walker, Map<String, Object> context)
-  {
-    return run(querySegmentSpec.lookup(this, walker), context);
-  }
-
   public Sequence<T> run(QueryRunner<T> runner, Map<String, Object> context)
   {
     return runner.run(this, context);
   }
 
-  @Override
-  public List<Interval> getIntervals()
+  public static Duration initDuration(QuerySegmentSpec querySegmentSpec)
```

Comment:
Method name is unclear, I don't see connection with what this method actually does.